### PR TITLE
add app service plan

### PR DESCRIPTION
**Pull request by yungezz**
add app service plan module from upstream

### CODE REVIEW

File: library/azure_rm_appserviceplan.py [hunk 1]
```diff
@@ -0,0 +1,354 @@
...
+
+    location:
+        description:
+            - Resource location. If not set, location from the resource group will be used as default.
+
+    sku:
+        description:
+            - The pricing tiers, e.g., F1, D1, B1, B2, B3, S1, P1, P1V2 etc.
+            - Please see U(https://azure.microsoft.com/en-us/pricing/details/app-service/plans/) for more detail.
+            - For linux app service plan, please see U(https://azure.microsoft.com/en-us/pricing/details/app-service/linux/) for more detail.
```
Comment (zikalino) [inline library/azure_rm_appserviceplan.py]: Linux uppercase

File: library/azure_rm_appserviceplan.py [hunk 2]
```diff
@@ -0,0 +1,354 @@
...
+
+        self.sku = None
+        self.is_linux = None
+        self.number_of_workers = 1
+
+        self.tags = None
+
+        self.results = dict(
+            changed=False,
+            ansible_facts=dict(azure_appserviceplan=None)
```
Comment (zikalino) [inline library/azure_rm_appserviceplan.py]: ansible_facts?

File: library/azure_rm_appserviceplan.py (hunk 1 above)
Comment (yungezz) [reply library/azure_rm_appserviceplan.py]: this is from upstream, will fix upstream, and sync from there

File: library/azure_rm_appserviceplan.py [hunk 3]
```diff
@@ -0,0 +1,354 @@
...
+    - name: update sku of existing windows app service plan
+      azure_rm_appserviceplan:
+        name: "windowsplan2"
+        resource_group: "appserviceplan_rg"
+        location: "eastus"
+        sku: S2
+'''
+
+RETURN = '''
+azure_appserviceplan:
```
Comment (zikalino) [inline library/azure_rm_appserviceplan.py]: I think id is just at the top of results?

File: library/azure_rm_appserviceplan.py (hunk 2 above)
Comment (yungezz) [reply library/azure_rm_appserviceplan.py]: this is from upstream,  will fix upstream, then sync to role

**Pull request merged**